library/graphs/mst.hpp

```diff
@@ -0,0 +1,34 @@
+/** @file */
+#pragma once
+#include "../../kactl/content/data-structures/UnionFind.h"
+/**
+ * Kruskal's algorithm to find a minimum spanning tree of a graph
+ *
+ * The implementation will find a minimum spanning forest if the graph is not
+ * connected
+ *
+ * @param eds a list of the edges of the form [u, v, w]
+ * @param n the number of vertices in the graph
+ * @returns [cost, ids] where `cost` is the total weight of the minimum spanning
+ * tree and `ids` is the list of the indices of the edges that are part of the
+ * minimum spanning tree
+ * @time O(n α(n) + m log m)
+ * @space O(n + m)
+ */
```

Comment:
also plz nit this, IDK if this looks quite right (aesthetically feels a bit off)